components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java

```diff
@@ -559,6 +611,14 @@ private static class SQLQuery {
                 "FROM IDN_IDENTITY_USER_DATA " +
                 "WHERE DATA_KEY = ? AND DATA_VALUE LIKE ? AND TENANT_ID = ? AND USER_NAME LIKE ?";
 
+        public static final String FILTER_USERS_BY_DATA_KEY_LESS_THAN_DATA_VALUE =
+                "SELECT USER_NAME, DATA_VALUE FROM IDN_IDENTITY_USER_DATA WHERE " +
+                        "DATA_KEY = ? AND TENANT_ID = ? AND DATA_VALUE < ?";
+
+        public static final String FILTER_USERS_BY_DATA_KEY_LESS_THAN_AND_GREATER_THAN_DATA_VALUES =
+                "SELECT USER_NAME, DATA_VALUE FROM IDN_IDENTITY_USER_DATA WHERE " +
+                        "DATA_KEY = ? AND TENANT_ID = ? AND DATA_VALUE < ? AND DATA_VALUE > ?";
```

Comment:
Please double-check the order of the parameters. The start and end boundaries should be, 

startValue < DATA_VALUE < endValue       -->  
DATA_VALUE > startValue and DATA_VALUE < endValue    -->  
DATA_VALUE > ? and DATA_VALUE < ?

What I see here is

DATA_VALUE < startValue and DATA_VALUE > endValue  -->  endValue < DATA_VALUE < startValue

Reply:
Fixed with https://github.com/wso2-extensions/identity-governance/pull/726/commits/41e2a3f5a5f65efc1f3d48b4fdece020ae3ece6b